stdlib/heap.c

```diff
@@ -118,7 +142,7 @@ void *__realloc(void *m, size_t size)
     else
     {
         void *n = __malloc(size);
-        __memcpy8(n, m, size / 8);
+        __memcpy8(n, m, (size + 7) / 8);
```

Comment:
Why is this necessary? This was not in the previous version of your PR.

Reply:
Another bug fix in the heap implementation

Reply:
Changed again!